providers/dns/hyperone/internal/client.go

```diff
@@ -77,9 +76,12 @@ func NewClient(apiEndpoint, locationID string, passport *Passport) (*Client, err
 // https://api.hyperone.com/v2/docs#operation/dns_project_zone_recordset_list
 func (c *Client) FindRecordset(zoneID, recordType, name string) (*Recordset, error) {
 	// https://api.hyperone.com/v2/dns/{locationId}/project/{projectId}/zone/{zoneId}/recordset
-	resourceURL := path.Join("dns", c.locationID, "project", c.projectID, "zone", zoneID, "recordset")
+	endpoint, err := url.JoinPath(c.apiEndpoint, "dns", c.locationID, "project", c.projectID, "zone", zoneID, "recordset")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
There's a lot of repetition here, in `CreateRecordset`, in `DeleteRecordset`, in `GetRecords`, and in `GetZones`.

Can we make `(*Client).apiEndpoint` a `*url.URL`, and use `c.apiEndoint.JoinPath(...)`?

Reply:
something like that? https://github.com/go-acme/lego/pull/1835/commits/b0b91728c49a6cb396fbfe8729db5ea474350e3e

Reply:
That, plus

```diff
-endpoint, err := url.JoinPath(c.apiEndpoint, "dns", c.locationID, "project", c.projectID, "zone", zoneID, "recordset")
-if err != nil {
-	return nil, err
-}
+endpoint := c.apiEndpoint.JoinPath("dns", c.locationID, "project", c.projectID, "zone", zoneID, "recordset")
```

Reply:
is it better like that? https://github.com/go-acme/lego/pull/1835/commits/5fd749110ab553bda056b1b81a4991c6e599c907

Reply:
clean the constructor in https://github.com/go-acme/lego/pull/1835/commits/0d78446553de4c048ab3366cb1d8a4d9cc1b96fb